serializer/string.go

```diff
@@ -0,0 +1,30 @@
+package serializer
+
+// NewString constructs a new string processer for the given stream
+func NewString(stream Stream) String {
+	s := String{
+		Byte: NewByte(stream),
+	}
+
+	return s
+}
+
+// String Read, Writer implementation of the string primitive.
+type String struct {
+	Byte
+}
+
+// Write writes the given byte buffer and the expecting message length to the underlaying data stream.
+func (s *String) Write(i string) error {
```

Comment:
What's the advantage of attaching these methods to a struct (`*serializer.String`) per type, over a single struct that wraps an `io.Writer` and has a method to serialize each primitive type, or just a method like `writeString(io.Writer, string)` with no struct involved?

Reply:
I wanted to create simple structs that could be extended by other structs. The string implementation is a good example where I extend the Byte functionality and interpret the returned bytes as a string.

Changing the structs into methods could be a good alternative. I will create a proposal somewhere in the next day.